### PR TITLE
added middleware for auth0

### PR DESCRIPTION
**Pull request by JordonP123**
# Description

In order to allow users to authenticate, we needed our API endpoints to be validated.

- What work was done?
  - Added new middleware for Auth0 that will create a profile if does not exist. 
  - Updated routes with new middleware.
- What feature / user story is it for?
  - [Jira ticket](https://bloomtechlabs.atlassian.net/jira/software/c/projects/BL/boards/7?modal=detail&selectedIssue=BL-1075)

- [LOOM](https://www.loom.com/share/3a3dd4f4fc2e4d418670d6648c464145)

### CODE REVIEW

File: api/middleware/authProfile.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+const createError = require('http-errors');
+const Profiles = require('../profile/profileModel');
+// const jwt = require('express-jwt');
```
Comment (tinomen) [inline api/middleware/authProfile.js]: Do we need this commented-out code? if not, lets remove it.

**Review by tinomen — CHANGES_REQUESTED** [review]
This is nice and small, well done.

File: api/middleware/authProfile.js (hunk 1 above)
Comment (Kobramall) [inline api/middleware/authProfile.js]: yeah just delete this. besides that it looks good to me!

Comment (JordonP123) [reply api/middleware/authProfile.js]: removed it!

**Review by JackNunnink — APPROVED** [review]

**Pull request closed without merging**